components/movementsensor/gpsrtkserial/gpsrtkserial_test.go

```diff
@@ -153,16 +159,17 @@ func TestPosition(t *testing.T) {
 	// If there is no last error, invalid current position and valid last position, return last position
 	t.Run("invalid position with valid last position, with position error", func(t *testing.T) {
 		g := &rtkSerial{
-			err:          movementsensor.NewLastError(1, 1),
-			lastposition: movementsensor.NewLastPosition(),
-			cachedData:   gpsutils.NewCachedData(&mockDataReader{}, logging.NewTestLogger(t)),
+			err:        movementsensor.NewLastError(1, 1),
+			cachedData: gpsutils.NewCachedData(&mockDataReader{}, logging.NewTestLogger(t)),
 		}
+		expectedPos := initializePosition(g.cachedData)
 
-		g.lastposition.SetLastPosition(testLastPosition)
-		expectedPos := geo.NewPoint(42.1, 123.)
+		// This is an invalid command, containing too many periods and colons.
+		invalidPosition := "$GPGLL,87.65.4321,N,987.65.4321,W,12:34:56.78,A,D*7F"
+		g.cachedData.ParseAndUpdate(invalidPosition)
```

Comment:
I added this part in to match the comments at the top of the test. Without this, I don't see anything in this test about an invalid current position! :flushed: